audobject/core/utils.py

```diff
@@ -119,6 +157,29 @@ def get_version(module_name: str) -> typing.Optional[str]:
         return None
 
 
+def install_package(
```

Comment:
As this function might be useful for others, should we move it to `audeer`?

Reply:
Yes, why not.

Reply:
So would you want to add it to `audeer` before we continue here or sometime later?

Reply:
Yes, I would first add it to `audeer` as I don't think we have time pressure here.

Reply:
Ok, I thought we wanted to have this feature before we complete the how-to-install-model instructions? :)

Reply:
You are referring to https://github.com/audeering/w2v2-how-to?
For that it is not needed as we don't need a feature extractor.

I would like to have it for https://gitlab.audeering.com/models/models.pp.audeering.com/-/merge_requests/21

Reply:
I was referring to the second link, cause that was my motivation to start with the PR.

Reply:
OK, but then it's fine. I'm pretty sure we will be able to work on it during the next days/weeks.

Reply:
Created https://github.com/audeering/audeer/pull/52

Reply:
I now switched to `audeer.install_package()`. At the moment it installs a package only if no version is installed yet. So should we raise an error if a different version than the desired version is installed?

Reply:
I would suggest to raise an error if a different version is present as we cannot know if it will work.

Reply:
The problem I see is that then `auto_install=True` has two effects. The obvious purpose is to install missing packages. But in addition it now serves as some sort of package sanity check. E.g. even if all packages are present, code that runs with `auto_install=False` may no longer work. So I wonder if we should handle the problem of mismatching package versions in some other way, e.g. another argument? And maybe we should also offer a way to control how strict the checks should be, e.g. do not break if a newer version of a package is installed.

Reply:
I'm not very much in favor of adding an additional argument to handle the error messages.
Even of `auto_install=False` works, you might still get wrong results because you used a different version of the package.

If you think raising an error is too strong we could also raise a warning instead. This could also be done in both cases, `auto_install=False` and `auto_install=True`.

Reply:
OK, then let's do that in a separate PR and stay with the current approach here, namely install a package if no version is installed yet or otherwise do nothing.

Reply:
OK